zapcore/encoder.go

```diff
@@ -112,21 +112,34 @@ func EpochNanosTimeEncoder(t time.Time, enc PrimitiveArrayEncoder) {
 	enc.AppendInt64(t.UnixNano())
 }
 
+type appendTimeEncoder interface {
```

Comment:
This will become part of the contract for custom encoders. Can we export this?
Or we can document on the ISO8601, RFC3339, and RFC3339Nano time encoders that
if the encoder implements this interface, we will use that method.